lib/liquid/block_body.rb

```diff
@@ -13,8 +14,46 @@ def initialize
       @blank = true
     end
 
-    def parse(tokenizer, parse_context)
+    def parse(tokenizer, parse_context, &block)
       parse_context.line_number = tokenizer.line_number
+
+      if tokenizer.for_liquid_tag
+        parse_for_liquid_tag(tokenizer, parse_context, &block)
+      else
+        parse_for_document(tokenizer, parse_context, &block)
+      end
+    end
+
+    private def parse_for_liquid_tag(tokenizer, parse_context)
+      while token = tokenizer.shift
+        case
+        when token.empty?
+          # pass
+        else
```

Comment:
why a `case` statement instead of replacing the above 4 lines with `unless token.empty?`

Reply:
We need to run the `parse_context.line_number =` statement below (since the line number can change even though the token is empty, unlike the regular tokenizer), so we can't use the `next unless token.empty?` pattern used in `parse_for_document`. This should definitely be made clearer in the comment.

I also just noticed the line number will be wrong if the token is just whitespace, because it hits the `next` statement on line 34.

Reply:
I didn't mean using `next`.  I was suggesting.

```diff
diff --git a/lib/liquid/block_body.rb b/lib/liquid/block_body.rb
index 4861bca..abeb591 100644
--- a/lib/liquid/block_body.rb
+++ b/lib/liquid/block_body.rb
@@ -26,11 +26,7 @@ module Liquid
 
     private def parse_for_liquid_tag(tokenizer, parse_context)
       while token = tokenizer.shift
-        case
-        when token.empty? || token =~ WhitespaceOrNothing
-          # pass, but assign line_number below, since it could change even when
-          # the token is empty
-        else
+        unless token.empty? || token =~ WhitespaceOrNothing
           unless token =~ LiquidTagToken
             # line isn't empty but didn't match tag syntax, yield and let the
             # caller raise a syntax error
```

Reply:
TIL that syntax! 👍 definitely